### PR TITLE
[Internal] When load data for batch run, only read first max_rows_count of  rows.

### PR DESCRIPTION
**Pull request by riddlexu**
# Description

When load data for batch run, we will load all the data of file in memory (do truncation after file loaded) even if max_rows_count is set.
This PR pass the max_rows_count to the file loader to speed up loading.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_utils/load_data.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def _handle_dir(dir_path: str, max_rows_count: int, logger: logging.Logger = Non
     while len(target_dir) > 0:
         files, dirs = _bfs_dir(target_dir)
         for file in files:
-            current_df = _pd_read_file(file, logger=logger)
+            current_df = _pd_read_file(file, logger=logger, max_rows_count=max_rows_count)
             df = pd.concat([df, current_df])
```
Comment (PeiwenGaoMS) [inline src/promptflow/promptflow/_utils/load_data.py:83]: Please also add a check here to see if it exceeds the `max_rows_count`.

File: src/promptflow/promptflow/batch/_batch_inputs_processor.py [hunk 2]
```diff
@@ -51,12 +51,22 @@ def _resolve_input_data(self, input_dirs: Dict[str, str]):
     def _resolve_data_from_input_path(self, input_path: Path):
         """Resolve input data from directory"""
         result = []
+        max_rows_count: int = None
+        if self._max_lines_count:
+            # Add one more row to check if the data exceeds the limit.
+            max_rows_count = self._max_lines_count + 1
```
Comment (PeiwenGaoMS) [inline src/promptflow/promptflow/batch/_batch_inputs_processor.py:57]: You can modify the content of the log and change '>' to '>='

**Comment by PeiwenGaoMS** [comment]
Please add `nrows` to the `ignoreWords` of `.cspell.json`
image

File: src/promptflow/promptflow/_utils/load_data.py (hunk 1 above)
Comment (riddlexu) [reply src/promptflow/promptflow/_utils/load_data.py:83]: Fixed. Thanks.

File: src/promptflow/promptflow/batch/_batch_inputs_processor.py (hunk 2 above)
Comment (riddlexu) [reply src/promptflow/promptflow/batch/_batch_inputs_processor.py]: Fixed. Thanks.

File: src/promptflow/promptflow/batch/_batch_inputs_processor.py [hunk 3]
```diff
@@ -51,15 +51,24 @@ def _resolve_input_data(self, input_dirs: Dict[str, str]):
...
         """Resolve input data from directory"""
         result = []
+        max_rows_count: int = None
+        if self._max_lines_count:
+            max_rows_count = self._max_lines_count
         if input_path.is_file():
-            result.extend(resolve_multimedia_data_recursively(input_path.parent, load_data(input_path)))
+            result.extend(resolve_multimedia_data_recursively(
+                input_path.parent,
+                load_data(local_path=input_path, max_rows_count=max_rows_count))
```
Comment (PeiwenGaoMS) [inline src/promptflow/promptflow/batch/_batch_inputs_processor.py:60]: Pass `self._max_lines_count` directly？

**Review by PeiwenGaoMS — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**